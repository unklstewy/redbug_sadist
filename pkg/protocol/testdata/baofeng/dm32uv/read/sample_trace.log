12:34:56.123456 write(3, "\x52\x01\x02\x03", 4) = 4
12:34:56.123500 read(3, "\x06", 1) = 1
12:34:56.124000 write(3, "\x57\x03\x04\x05", 4) = 4
12:34:56.124100 read(3, "\x06", 1) = 1
12:34:56.124200 write(3, "\x50\x10\x20\x30", 4) = 4
12:34:56.124300 read(3, "\x02\x41\x42\x43", 4) = 4
12:34:56.124400 write(3, "PROG", 4) = 4
12:34:56.124500 read(3, "\x15", 1) = 1
