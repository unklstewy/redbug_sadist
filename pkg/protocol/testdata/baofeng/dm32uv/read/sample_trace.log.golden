[
  {
    "Timestamp": "12:34:56.123456",
    "Direction": "PC→Radio",
    "FileDesc": "3",
    "RawHex": "52010203",
    "DecodedASCII": "R...",
    "Length": 4,
    "CommandType": "Read Request",
    "Notes": ""
  },
  {
    "Timestamp": "12:34:56.123500",
    "Direction": "Radio→PC",
    "FileDesc": "3",
    "RawHex": "06",
    "DecodedASCII": ".",
    "Length": 1,
    "CommandType": "ACK (Acknowledge)",
    "Notes": ""
  },
  {
    "Timestamp": "12:34:56.124000",
    "Direction": "PC→Radio",
    "FileDesc": "3",
    "RawHex": "57030405",
    "DecodedASCII": "W...",
    "Length": 4,
    "CommandType": "Write Request",
    "Notes": ""
  },
  {
    "Timestamp": "12:34:56.124100",
    "Direction": "Radio→PC",
    "FileDesc": "3",
    "RawHex": "06",
    "DecodedASCII": ".",
    "Length": 1,
    "CommandType": "ACK (Acknowledge)",
    "Notes": ""
  },
  {
    "Timestamp": "12:34:56.124200",
    "Direction": "PC→Radio",
    "FileDesc": "3",
    "RawHex": "50102030",
    "DecodedASCII": "P. 0",
    "Length": 4,
    "CommandType": "Program Command",
    "Notes": ""
  },
  {
    "Timestamp": "12:34:56.124300",
    "Direction": "Radio→PC",
    "FileDesc": "3",
    "RawHex": "02414243",
    "DecodedASCII": ".ABC",
    "Length": 4,
    "CommandType": "Data Packet",
    "Notes": ""
  },
  {
    "Timestamp": "12:34:56.124500",
    "Direction": "Radio→PC",
    "FileDesc": "3",
    "RawHex": "15",
    "DecodedASCII": ".",
    "Length": 1,
    "CommandType": "NAK (Negative Acknowledge)",
    "Notes": ""
  }
]